src/esm_runscripts/filedicts.py

```diff
@@ -73,6 +74,78 @@ def inner_method(self, *args, **kwargs):
     return inner_method
 
 
+def globbing(method):
+    """
+    Decorator method for ``SimulationFile``'s methods ``cp``, ``mv``, ``ln``, that
+    enables globbing. If a ``*`` is found on the ``source`` or ``target`` the globbing
+    logic is activated, and consist of:
+    - run checks for globbing syntax
+    - check if any file matches the globbing pattern
+    - construct one instance of ``SimulationFile`` for each file matching the globbing
+    - run the ``method`` for that particular file
+
+    Parameters
+    ----------
+    method : method
+        The decorated method (``cp``, ``mv``, ``ln``)
+
+    Returns
+    -------
+    method : method
+        If no globbing is needed, returns the method as it was given originally.
+    """
+    @functools.wraps(method)
+    def inner_method(self, source, target, *args, **kwargs):
+        method_name = method.__name__
+        source_name = self[f"name_in_{source}"]
+        target_name = self[f"name_in_{target}"]
+
+        if "*" in source_name or "*" in target_name:
+            # Get wildcard patterns
+            source_pattern = source_name.split("*")
+            target_pattern = target_name.split("*")
+
+            # Check wild cards syntax
+            self.wild_card_check(source_pattern, target_pattern)
+
+            # Obtain source files
+            glob_source_paths = self.find_globbing_files(source)
+
+            # Extract globbing source names
+            glob_source_names = []
+            for glob_source_path in glob_source_paths:
+                glob_source_names.append(pathlib.Path(glob_source_path).name)
+
+            # Solve the globbing target names
+            glob_target_names = []
+            for glob_source_name in glob_source_names:
+                glob_target_name = glob_source_name
+                for sp, tp in zip(source_pattern, target_pattern):
+                    glob_target_name = glob_target_name.replace(sp, tp)
+                glob_target_names.append(glob_target_name)
+
+            # Loop through source files
+            for glob_source_name, glob_target_name in zip(
+                glob_source_names, glob_target_names
+            ):
+                # Create a new simulation file object for this specific glob file
+                glob_config = copy.deepcopy(self._config)
+                glob_dict = dpath.util.get(
+                    glob_config, self.attrs_address, separator=".", default={}
+                )
+                glob_dict[f"name_in_{source}"] = glob_source_name
+                glob_dict[f"name_in_{target}"] = glob_target_name
+                glob_file = SimulationFile(glob_config, self.attrs_address)
+                # Use method
+                this_method = getattr(glob_file, method_name)
+                this_method(source, target, *args, **kwargs)
+        else:
+            return method(self, source, target, *args, **kwargs)
```

Comment:
You could save yourself a bit of code here (double check the indents, GitHub's online review suggest thingy isn't good about that):
```suggestion
                # Use method
                method = getattr(glob_file, method_name)
        return method(self, source, target, *args, **kwargs)
```

Reply:
Good point!